focuspoints.lrdevplugin/DefaultPointRenderer.lua

```diff
@@ -30,104 +30,127 @@ require "ExifUtils"
   log("Shot orientation: " .. shotOrientation)
 
-  local originalX, originalY = DefaultPointRenderer.funcGetAFPixels(targetPhoto, metaData)
-
-  log( "cL: " .. cropLeft .. ", cT: " .. cropTop .. ", cAngle: " .. math.deg(cropAngle) .. "°")
-
-  local x = originalX
-  local y = originalY
-
-  --[[ lightroom does not report if a photo has been rotated. code below
-        make sure the rotation matches the expected width and height --]]
-  local additionalRotation = 0
-  if shotOrientation == 90 then
-    x = originalY
-    y = originalHeight - originalX
-    cropLeft = developSettings["CropTop"]
-    cropTop = 1 - developSettings["CropRight"]
-    additionalRotation = math.pi / 2
-  elseif shotOrientation == 270 then
-    x = originalWidth - originalY
-    y = originalX
-    cropLeft = 1 - developSettings["CropBottom"]
-    cropTop = developSettings["CropLeft"]
-    additionalRotation = -math.pi / 2
+  local pointsTable = DefaultPointRenderer.funcGetAfPoints(photo, metaData)
+  if pointsTable == nil then
+    return nil
   end
 
-  log("shotOrientation: " .. shotOrientation .. "°, totalRotation: " .. math.deg(cropAngle + additionalRotation) .. "°")
-
-  -- Rotation around 0,0
-  local rX = x * math.cos(cropAngle) + y * math.sin(cropAngle)
-  local rY = -x * math.sin(cropAngle) + y * math.cos(cropAngle)
-
-  -- Rotation of top left corner
-  local oX = cropLeft * originalWidth
-  local oY = cropTop * originalHeight
-  local roX = oX * math.cos(cropAngle) + oY * math.sin(cropAngle)
-  local roY = -oX * math.sin(cropAngle) + oY * math.cos(cropAngle)
-
-  -- Translation so the top left corner become the origin
-  local tX = rX - roX
-  local tY = rY - roY
+  -- Looping through the af-points and drawing them depending on their nature
+  local viewsTable = {}
+  for key, point in pairs(pointsTable.points) do
+    local originalX = point.x
+    local originalY = point.y
+
+    local x = originalX
+    local y = originalY
+
+    --[[ lightroom does not report if a photo has been rotated. Code below
+          make sure the rotation matches the expected width and height --]]
+    local additionalRotation = 0
+    if shotOrientation == 90 then
+      x = originalY
+      y = originalHeight - originalX
+      cropLeft = developSettings["CropTop"]
+      cropTop = 1 - developSettings["CropRight"]
+      additionalRotation = math.pi / 2
+    elseif shotOrientation == 270 then
+      x = originalWidth - originalY
+      y = originalX
+      cropLeft = 1 - developSettings["CropBottom"]
+      cropTop = developSettings["CropLeft"]
+      additionalRotation = -math.pi / 2
+    end
+
+    log("shotOrientation: " .. shotOrientation .. "°, totalRotation: " .. math.deg(cropAngle + additionalRotation) .. "°")
+
+    local template = pointsTable.pointTemplates[point.pointType]
+    if template == nil then
+      LrErrors.throwUserError("Point template " .. point.pointType .. " could not be found.")
```

Comment:
throwing an error stops execution. The else statement is not necessary and in a language that's based upon indenting and no curly brackets, I have a high prefer to let it exit rather than wrapping in an else. 

To see for yourself try: 
```
log("About to throw an error")
LrErrors.throwUserError("My sample error throw. Will the next line log? ")
log("Will this log")
```
"Will this log" is never called and execution stops at the full.